src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -214,9 +232,25 @@ private void ReturnEntriesToPool()
 
 		internal IEnumerable<DependencyPropertyDetails> GetAllDetails() => Entries.Trim();
 
+		/// <summary>
+		/// Adds a default value provider.
+		/// </summary>
+		/// <param name="provider">Default value provider.</param>
+		/// <remarks>
+		/// Providers which are registered later have higher precedence.
+		/// E.g. when both a derived and base class register their own default
+		/// value provider in the constructor for the same property, the derived
+		/// class value is used.
+		/// </remarks>
 		public void RegisterDefaultValueProvider(DependencyObjectStore.DefaultValueProvider provider)
 		{
-			_defaultValueProvider = provider;
+			if (provider == null)
+			{
+				throw new ArgumentNullException(nameof(provider));
+			}
+
+			_defaultValueProviders ??= new List<DependencyObjectStore.DefaultValueProvider>();
```

Comment:
I think the default length of a list (usually 16) is too much.
```suggestion
			_defaultValueProviders ??= new List<DependencyObjectStore.DefaultValueProvider>(2);
```